src/Workspaces/Core/Portable/Workspace/Solution/Solution.CompilationTracker.cs

```diff
@@ -660,6 +657,30 @@ public CompilationInfo(Compilation compilation, bool hasCompleteReferences)
                 }
             }
 
+            private async Task<bool> ComputeDependentCompleteAsync(Solution solution, bool isComplete, CancellationToken cancellationToken)
+            {
+                if (!isComplete)
+                {
+                    return false;
+                }
+
+                foreach (var projectReference in this.ProjectState.ProjectReferences)
```

Comment:
Given we're already doing this in the other function which is also computing compute bits, why not just do it in the same loop?